### PR TITLE
Loosen unsatisfied constraints for npm packages

### PR DESCRIPTION
**Pull request by ybiquitous**
> Explain a summary, purpose, or background for this change.

This change aims to loosen the behavior when npm-installed packages are unsatisfied with our version constraints.

**Before**: an analysis error is raised
⬇️ 
**After**: fall back to the default (pre-installed) version with a warning

I think the change is reasonable because raising an error is too strict.

Also, this change does not use the `npm ls` command because:
- it will be changed with npm v7 (see also #1963)
- it is hard to control

Instead of `npm ls`, this checks `node_modules/{package_name}/` directories to get installed versions.

> Refer related issues or pull requests (e.g. `Close #123` or `None`).

Related to #1963 

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

File: test/nodejs_test.rb [hunk 1]
```diff
@@ -236,70 +208,141 @@ def test_install_nodejs_deps_with_option_none_and_without_package_json
...
+    end
+  end
+
+  def test_install_nodejs_deps_with_missing_deps
+    with_workspace do |workspace|
+      new_processor(workspace: workspace)
 
-        stdout, _ = processor.capture3!(processor.nodejs_analyzer_bin, "-v")
-        assert_equal "v6.0.1\n", stdout
+      processor.package_json_path.write({ dependencies: { "classcat": "5.0.3" }}.to_json)
```
Comment (ybiquitous) [inline test/nodejs_test.rb:309]:
```suggestion
      processor.package_json_path.write({ dependencies: { "classcat" => "5.0.3" }}.to_json)
```

File: test/nodejs_test.rb [hunk 2]
```diff
@@ -236,70 +208,141 @@ def test_install_nodejs_deps_with_option_none_and_without_package_json
...
+      assert_includes actual_messages, "All constraints are not satisfied. The default version `1.0.1` will be used instead."
+      assert_equal "1.0.1", processor.analyzer_version
+    end
+  end
+
+  def test_install_nodejs_deps_with_unsatisfied_deps
+    with_workspace do |workspace|
+      new_processor(workspace: workspace)
+
+      processor.package_json_path.write({ dependencies: { "classcat": "5.0.3", "is-string": "1.0.5" }}.to_json)
```
Comment (ybiquitous) [inline test/nodejs_test.rb:329]: > __[RuboCop]__ Unnecessary symbol conversion; use `classcat:` instead. ([view](https://sider.review/gh/repos/202486306/pulls/2171#issue-5437032))
>
> <details>
>
> | Rule | Severity |
> | ---- | -------- |
> | `Lint/SymbolConversion` | `warning` |
>
> References:
> - <https://docs.rubocop.org/rubocop/cops_lint.html#lintsymbolconversion>
>
> You can [close this issue](https://sider.review/gh/repos/202486306/pulls/2171#issue-5437032) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

Fixing...

Comment (ybiquitous) [reply test/nodejs_test.rb:329]:
```suggestion
      processor.package_json_path.write({ dependencies: { "classcat" => "5.0.3", "is-string" => "1.0.5" }}.to_json)
```

**Review by yykamei — APPROVED** [review]
LGTM

**Comment by ybiquitous** [comment]
@yykamei Thanks for the review. I'll exclude it from the next version of Runners because this includes a breaking change.

**Pull request merged**